### DISCUSSION TITLE
What does "Health Check Failed" mean?

### DISCUSSION THREAD

**Discussion by shadow7412** [teslamate-org/teslamate#1103] [Q&A]
![image](https://user-images.githubusercontent.com/286341/100570650-56eb4d80-3325-11eb-888c-db10073b7bc7.png)

I can't see anything in the car itself implying something is wrong, and there's nothing coming up on the Tesla app. I don't see anything about this warning in the teslamate docs.

What does it mean? How can I find out what it thinks is wrong with my car?

**Comment by airmaxx** [comment]
To quote Adrian: The failed health check indicates that there is an issue with the Tesla servers.

**Comment by adriankumpf** [comment]
The error message is not related to your car, but to the communication with the Tesla servers. If the problem does not resolve itself after some time, take a look at the TeslaMate logs.

**Comment by jayali786** [comment]
I am getting the same issue so many error regarding tesla api communication,
i also did fresh install same error over and over, it is like every day i have to restart my computer to fix this issue

**Comment by marstein** [comment]
It could maybe be a problem with resolving the DNS address of the tesla servers. I got

```
teslamate_1  | 2023-10-25 23:33:45.035 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/149444448996579 -> error: "non-existing domain" (8002.768 ms)
```

That solved itself after a while.

**Comment by damccull** [comment]
I've seen this a few times in the last few days since I installed Teslamate. When the servers are back again, will Teslamate pick up the events that happened while there was a communications issue or will those data be lost forever? Is teslafi likely to be experiencing the same thing?

**Reply by cwanja** [reply]
> When the servers are back again, will Teslamate pick up the events that happened while there was a communications issue or will those data be lost forever? 

No. There's no backfill feature. 

> Is teslafi likely to be experiencing the same thing?

Maybe. Assume they are using the owners API similar to TeslaMate and thus would experience the same issue. No clue how they handle it or even present it to the user to know.

**Comment by damccull** [comment]
Restarting the teslamate server seems to work. Does it have a self-recovery option? or is there an endpoint I can view to get the health check status and restart automatically if I detect it's offline like that?

**Reply by cwanja** [reply]
I personally believe your restart is a placebo effect more than a 'fix'. It recovers on its own and starts gathering data. When I have seen it, I leave TeslaMate alone and it recovers fine.

**Reply by damccull** [reply]
Ah, you might be right. It did come back on its own this time.
